packages/loot-core/src/server/filters/app.ts

```diff
@@ -0,0 +1,169 @@
+import * as uuid from '../../platform/uuid';
+import { parseConditionsOrActions } from '../accounts/transaction-rules';
+import { createApp } from '../app';
+import * as db from '../db';
+import { requiredFields } from '../models';
+import { mutator } from '../mutators';
+import { batchMessages } from '../sync';
+import { undoable } from '../undo';
+
+let app = createApp();
+
+export const filterModel = {
+  validate(filter, { update }: { update?: boolean } = {}) {
+    requiredFields('transaction_filters', filter, ['conditions'], update);
+
+    if (!update || 'conditionsOp' in filter) {
+      if (!['and', 'or'].includes(filter.conditionsOp)) {
+        throw new Error('Invalid filter conditionsOp: ' + filter.conditionsOp);
+      }
+    }
+
+    return filter;
+  },
+
+  toJS(row) {
+    let { conditions, conditions_op, ...fields } = row;
+    return {
+      ...fields,
+      conditionsOp: conditions_op,
+      conditions: parseConditionsOrActions(conditions),
+    };
+  },
+
+  fromJS(filter) {
+    let { conditionsOp, ...row } = filter;
+    if (conditionsOp) {
+      row.conditions_op = conditionsOp;
+    }
+    return row;
+  },
+};
+
+export async function filterNameExists(name, filterId, newItem) {
+  let idForName = await db.first(
+    'SELECT id from transaction_filters WHERE tombstone = 0 AND name = ?',
+    [name],
+  );
+
+  if (idForName === null) {
+    return false;
+  }
+  if (!newItem) {
+    return idForName.id !== filterId;
+  }
+  return true;
+}
+
+//TODO: Possible to simplify this?
+//use filters and maps
+export function ConditionExists(item, filters, newItem) {
```

Comment:
All of our other functions start with lowercase letters, this should follow the pattern.

```suggestion
export function conditionExists(item, filters, newItem) {
```